icu4j/main/classes/core/src/com/ibm/icu/text/SimpleDateFormat.java

```diff
@@ -1944,7 +1959,7 @@ protected void subFormat(StringBuffer buf,
             if (toAppend == null) {
                 // Time isn't exactly midnight or noon (as displayed) or localized string doesn't
                 // exist for requested period. Fall back to am/pm instead.
-                subFormat(buf, 'a', count, beginOffset, fieldNum, capitalizationContext, pos, cal);
+                subFormat(buf, 'a', count, beginOffset, fieldNum, capitalizationContext, pos, 'b', cal);
```

Comment:
Add a comment pointing to the bug explaining why we have a different display field from data field.  (here and elsewhere)